aten/src/ATen/FunctionalTensorWrapper.cpp

```diff
@@ -537,6 +541,12 @@ bool isFunctionalTensor(ITensorListRef list) {
   return isFunctionalTensorIListRef(list);
 }
 
+void freeze_functional_tensor(const Tensor& tensor) {
+  TORCH_INTERNAL_ASSERT(at::functionalization::impl::isFunctionalTensor(tensor));
```

Comment:
Should this be a TORCH_CHECK since this is called from python and can be quite easily broken?

Reply:
it's not a public python api, and we should maintain the appropriate invariant python side

Reply:
Since we keep litigating this, I wonder if we should split internal assert into two variants, one for "this should be true invariant to c++ changes" and "this should be true invariant to python changes". Might be hard for ppl to understand